[quantization] Trivial quantized torch.mean implementation

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26253 [quantization] Trivial quantized torch.mean implementation**

Differential Revision: [D17529994](https://our.internmc.facebook.com/intern/diff/D17529994)

Commit:
[quantization] Trivial quantized torch.mean implementation

Commit:
Update on "[quantization] Trivial quantized torch.mean implementation"

Commit:
Update on "[quantization] Trivial quantized torch.mean implementation"




[ghstack-poisoned]

Commit:
Update on "[quantization] Trivial quantized torch.mean implementation"




[ghstack-poisoned]

Commit:
Update on "[WIP][quantization] Trivial quantized torch.mean implementation"

Commit:
Update on "[WIP][quantization] Trivial quantized torch.mean implementation"

Commit:
Update on "[WIP][quantization] Trivial quantized torch.mean implementation"




[ghstack-poisoned]

dzhulgakov reviewed (approved):
Namespaces are hard - but sure! :)

Want to add a comment somewhere on why the name is so strange?

Commit:
Update on "[WIP][quantization] Trivial quantized torch.mean implementation"


Differential Revision: [D17529994](https://our.internmc.facebook.com/intern/diff/D17529994)

Commit:
Update on "[WIP][quantization] Trivial quantized torch.mean implementation"


Differential Revision: [D17529994](https://our.internmc.facebook.com/intern/diff/D17529994)

Commit:
Update on "[WIP][quantization] Trivial quantized torch.mean implementation"


Differential Revision: [D17529994](https://our.internmc.facebook.com/intern/diff/D17529994)

[ghstack-poisoned]

Commit:
Update on "[quantization] Trivial quantized torch.mean implementation"


Differential Revision: [D17529994](https://our.internmc.facebook.com/intern/diff/D17529994)

zou3519 commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1646,23 +1646,43 @@
   variants: function, method
   supports_named_tensor: True
+  dispatch:
+    CPU: mean_cpu_gpu
```
This is wrong -- the named tensor version of mean is supposed to be a wrapper around regular mean.
